src/state-manager/TransactionQueue.ts

```diff
@@ -411,15 +435,63 @@ class TransactionQueue {
 
           for (let i = 0; i < req.stateList.length; i++) {
             // eslint-disable-next-line security/detect-object-injection
-            const state = req.stateList[i]
-            if (
-              i !== 0 &&
-              !this.validateCorrespondingTellSender(queueEntry, state.accountId, header.sender_id)
-            ) {
-              /* prettier-ignore */ if (logFlags.error && logFlags.verbose) this.mainLogger.error(`${route} validateCorrespondingTellSender failed for ${state.accountId}`)
-              return errorHandler(RequestErrorEnum.InvalidSender)
+            const state = req.stateList[i];
+            let isSenderValid = false
+            const senderNodeId = header.sender_id
+            if (configContext.p2p.useFactCorrespondingTell) {
+              let isSenderOurExeNeighbour = false
+              // check if it is a neighbour exe node sharing data
+              if (configContext.stateManager.shareCompleteData) {
+                const senderIsInExecutionGroup = queueEntry.executionGroupMap.has(senderNodeId)
+                const neighbourNodes = utils.selectNeighbors(queueEntry.executionGroup, queueEntry.ourExGroupIndex, 2) as Shardus.Node[]
+                const neighbourNodeIds = neighbourNodes.map((node) => node.id)
+                isSenderOurExeNeighbour = senderIsInExecutionGroup && neighbourNodeIds.includes(senderNodeId)
+                if (isSenderOurExeNeighbour) {
```

Comment:
all 4 above lines can be done outside of the loop of accounts.

Reply:
great catch, I have corrected locally for my next PR  (in the binary handler)